File: leetcode/randomProbs/islandPerimeter.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        int perimeter = 0;
        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[0].length; j++) {
                perimeter += helper(matrix, i, j);
            }
        }
        return perimeter;
    }
    
    public int helper(int[][] matrix, int x, int y) {
        int count = 0;
        if (matrix[x][y] == 1) {
            count = 4;
            if (x < matrix.length - 1 && matrix[x + 1][y] == 1) count--;
            if (x > 0 && matrix[x - 1][y] == 1) count--;
            if (y < matrix[0].length - 1 && matrix[x][y + 1] == 1) count--;
            if (y > 0 && matrix[x][y - 1] == 1) count--;
        }
        return count;
    }
}

File: leetcode/randomProbs/easy/islandPerimeter.py
class Solution:
    def helper(self, grid, i, j):
        count = 4
        if i-1 >= 0 and grid[i-1][j] == 1: count -= 1
        if i+1 < len(grid) and grid[i+1][j] == 1: count -= 1
        if j-1 >= 0 and grid[i][j-1] == 1: count -= 1
        if j+1 < len(grid[0]) and grid[i][j+1] == 1: count -= 1
        return count
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        ans = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    ans += self.helper(grid, i, j)
        return ans
